### PR TITLE
Add discussion Q&A template [skip ci]

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
### Description

Add a template for Q&A so users will provide required information for us to answer their questions

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: .github/DISCUSSION_TEMPLATE/q-a.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  - type: input
+    attributes:
+      label: NVFlare version (`python3 -m pip list | grep "nvflare"`)
+      description: NVFlare version (`python3 -m pip list | grep "nvflare"`)
+      placeholder: "2.3.8"
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: NVFlare branch (if running examples, please use the corresponding branch, `git branch`)
```
Comment (SYangster) [inline .github/DISCUSSION_TEMPLATE/q-a.yml:24]: Suggest reword to "if running examples, please use the branch that corresponds to the NVFlare version"?

Comment (YuanTingHsieh) [reply .github/DISCUSSION_TEMPLATE/q-a.yml]: thanks, reworded

File: .github/DISCUSSION_TEMPLATE/q-a.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      label: Operating system
+      description: Operating system (Ubuntu, MacOS, Windows, etc)
+      placeholder: "Ubuntu 20.04"
+    validations:
+      required: true
+  - type: checkboxes
+    attributes:
+      label: Have you successfully run the following examples?
+      options:
+        - label: hello-numpy-sag with simulator
```
Comment (holgerroth) [inline .github/DISCUSSION_TEMPLATE/q-a.yml:38]: Do we need to ask for all of them? Maybe split it up into several yes/no questions for each of the examples.

Comment (YuanTingHsieh) [reply .github/DISCUSSION_TEMPLATE/q-a.yml:38]: This is going to render as 4 check boxes that is not mandatory.

Do you prefer it to be ask four times yes/no checkboxes? (Not mandatory as well)

Comment (holgerroth) [reply .github/DISCUSSION_TEMPLATE/q-a.yml:38]: Ah. 4 checkboxes as now is fine. Just rephrase it as "Have you successfully run any of the following examples?"

Comment (YuanTingHsieh) [reply .github/DISCUSSION_TEMPLATE/q-a.yml:38]: Thanks will do!

**Review by holgerroth — APPROVED** [review]
LGTM

**Pull request merged**